src/main/java/com/epam/dataservice/AccidentBatchProcessor.java

```diff
@@ -0,0 +1,97 @@
+package com.epam.dataservice;
+
+import com.epam.data.RoadAccident;
+
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by bill on 16-5-1.
+ */
+public class AccidentBatchProcessor  implements Runnable {
+    private static AtomicInteger serialNo = new AtomicInteger();
+// For record Enrichment
+    private static PoliceForceService policeForceService = new PoliceForceService();
+    private static ExecutorService enrichExecutor = Executors.newFixedThreadPool(HomeWork3.MAX_ENRICH_THREADS);
+    private CompletionService<RoadAccident> enrichCompletionService = new ExecutorCompletionService<RoadAccident>(enrichExecutor);
+
+    private String taskName = getClass().getSimpleName() + serialNo.incrementAndGet();
+    private BlockingQueue<List<RoadAccident>> readQueue;
+    private List<BlockingQueue<RoadAccident>> writeQueues;
+
+    public  AccidentBatchProcessor(BlockingQueue<List<RoadAccident>> readQueue, List<BlockingQueue<RoadAccident>> writeQueues){
+        this.readQueue = readQueue;
+        this.writeQueues = writeQueues;
+    }
+
+    @Override
+    public void run() {
+        int dataCounter = 0;
+        List<RoadAccident> consumedData = null;
+        Boolean done = false;
+        try {
+            while (!done) {
+                consumedData = readQueue.take();
+                if (consumedData != null ) {
+                    if(!consumedData.isEmpty()) {
```

Comment:
It's not really a nice code when you have 4 levels of try-while-if-if. 
I understand that you did already a lot of cleanign, so not asking you to change this part. But I really want to you to try avoid so many layers in your code in future. Create some methods, combine conditions, etc. 

Try to make it prettier, ok? :smirk: